packages/vite/src/node/plugins/dynamicImportPolyfill.ts

```diff
@@ -111,8 +109,9 @@ function polyfill(modulePath = '.', importFunctionName = '__import__') {
         const absURL = new URL(url, baseURL)
 
         // If the module has already been imported, resolve immediately.
-        if (self[importFunctionName].moduleMap[absURL]) {
-          return resolve(self[importFunctionName].moduleMap[absURL])
+        const moduleFromUrl = self[importFunctionName].moduleMap[absURL as any]
```

Comment:
For now I used `as any` but potentially we could (in a later PR) use a Map instead :thinking: 
I just don't know if this code needs to run in environments where es5 is needed... but it uses const :thinking: so could work